### PR TITLE
Add Mainline Mensware article

### PR DESCRIPTION
**Pull request by tomayac**
Changes proposed in this pull request:

- Add Mainline Mensware article.

### CODE REVIEW

File: src/site/content/en/blog/mainline-mensware/index.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        purgeOnQuotaError: true,
+      }),
+    ],
+  }),
+);
+```
+
+{% Aside 'objective' %} Even though the caching strategy is exactly the same as the previous one, by
+splitting images into two caches (`product_images` and `images`), it allows for more flexible
+updates to the strategies or caches to future versions of the service worker. {% endAside %}
```
Comment (jpmedley) [inline src/site/content/en/blog/mainline-mensware/index.md:263]:
```suggestion
updates to the strategies or caches. {% endAside %}
```

File: src/site/content/en/blog/mainline-mensware/index.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+```js
+const OFFLINE_HTML = '/offline/offline.html';
+const PRECACHE = [
+  { url: OFFLINE_HTML, revision: '70f044fda3e9647a98f084763ae2c32a' },
+  { url: '/offline/offline.svg', revision: 'efe016c546d7ba9f20aefc0afa9fc74a' },
+];
+```
+
+The precache list is then fed into Workbox which takes care of all the heavy lifting of adding the
+URLs to the cache, checking for any revision mismatch, updating accordingly, and serving the
```
Comment (jpmedley) [inline src/site/content/en/blog/mainline-mensware/index.md:279]:
```suggestion
URLs to the cache, checking for any revision mismatch, updating, and serving the
```

File: src/site/content/en/blog/mainline-mensware/index.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+## Demo
+
+<figure class="w-figure">
+  {% Video src="video/8WbTDNrhLsU0El80frMBGE4eMCD3/eJgApjFLpSRFMcMyC4e0.mp4", width="300", autoplay=true, loop=true, muted=true %}
+  <figcaption class="w-figcaption">Offline page example as seen on www.mainlinemenswear.co.uk.</figcaption>
+</figure>
+
+### Reporting successful installs
+
+Apart from the homescreen launch tracking (with `"start_url": "/?utm_source=pwa"` in the web
```
Comment (jpmedley) [inline src/site/content/en/blog/mainline-mensware/index.md:318]:
```suggestion
Apart from the home screen launch tracking (with `"start_url": "/?utm_source=pwa"` in the web
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: src/site/content/en/blog/mainline-mensware/index.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+{% Aside %} See [Add a web app manifest](/add-manifest/) for a more in-depth explanation of all the
+web application manifest fields. {% endAside %}
+
+### Runtime caching for faster navigations
+
+Caching for web apps is a "must" for page speed optimization and for providing a better user
+experience for returning users.
+
+For caching on the web, there are quite a few strategies out there with each using
+[different approaches](https://dev.to/jonchen/service-worker-caching-and-http-caching-p82). The team
```
Comment (tomayac) [inline src/site/content/en/blog/mainline-mensware/index.md:142]:
```suggestion
For caching on the web, there are quite a few
[different approaches](https://dev.to/jonchen/service-worker-caching-and-http-caching-p82). The team
```

[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/mainline-mensware/index.md:142]: Yes, I have shortened this.

File: src/site/content/en/blog/mainline-mensware/index.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+maintain, [Workbox](https://developers.google.com/web/tools/workbox) has provided them with an easy
+way of declaring such complex strategies and has eased the pain of maintenance.
+
+#### Caching CSS and JS
+
+For CSS and JS files, the team has chosen to cache them and serve them over the cache using the
+[`StaleWhileRevalidate`](https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-strategies.StaleWhileRevalidate)
+Workbox strategy. This strategy allows them to serve all Nuxt CSS and JS files fast, increasing
+their page speed, while the files are being updated in the background for a fresh version of the
+files the next time they are being requested:
```
Comment (tomayac) [inline src/site/content/en/blog/mainline-mensware/index.md:158]:
```suggestion
Workbox strategy. This strategy allows them to serve all Nuxt CSS and JS files fast,
which significantly increases their site's performance.
At the same time, the files are being updated in the background to the latest version for the next visit:
```

[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/mainline-mensware/index.md:158]: Split in two and redundancy reduced.

**Comment by tomayac** [comment]
Thanks for the review, @jpmedley! PTAL.

File: src/site/content/en/blog/mainline-mensware/index.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+```js
+workbox.precaching.precacheAndRoute(PRECACHE);
+```
+
+#### Handling offline navigations
+
+Once the service worker activates and the offline page is precached, it is then used to **respond to
+offline navigation requests by the user**. While Mainline Menswear's web app is an SPA, the offline
+page shows only after the page reloads, the user closes and reopens the browser tab, or when the web
+app is launched from the homescreen while offline.
```
Comment (jpmedley) [inline src/site/content/en/blog/mainline-mensware/index.md:291]:
```suggestion
app is launched from the home screen while offline.
```

Comment (jpmedley) [reply src/site/content/en/blog/mainline-mensware/index.md:291]: @tomayac Sorry for not finding this yesterday.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**